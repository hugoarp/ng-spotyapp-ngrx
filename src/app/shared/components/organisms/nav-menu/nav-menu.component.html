<nav class="nav-menu">
  <img
    src="../../../../../assets/img/spotify-logo.svg"
    alt="Logo de spotify"
    class="nav-menu__logo"
  />
  <section class="nav-menu__pages">
    <app-nav-button
      *ngFor="let item of menuItems"
      [icon]="item.icon"
      [label]="item.label"
      [routerLink]="item.route"
    ></app-nav-button>
  </section>
  <app-user-button
    [user]="currentUser"
    class="nav-menu__user"
  ></app-user-button>
</nav>
