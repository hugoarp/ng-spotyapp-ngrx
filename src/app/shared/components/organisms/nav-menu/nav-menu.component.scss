@import "variables";
@import "mixins";

.nav-menu {
  width: 100%;
  background-color: black;
  padding: 0.5rem 0;
  display: flex;
  justify-content: space-evenly;
  position: fixed;
  bottom: 0;
  @include from($medium) {
    bottom: auto;
    left: 0;
    top: 0;
    flex-direction: column;
    width: 10rem;
    height: 100%;
    padding: 1rem;
    justify-content: flex-start;
    gap: 1.5rem;
  }
}
