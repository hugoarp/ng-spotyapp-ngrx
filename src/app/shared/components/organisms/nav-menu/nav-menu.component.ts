import { Component, OnInit, Input } from '@angular/core';
import { UserInfo } from 'src/app/core/services/interfaces/user';
import { SpotifyService } from 'src/app/core/services/spotify.service';

@Component({
  selector: 'app-nav-menu',
  templateUrl: './nav-menu.component.html',
  styleUrls: ['./nav-menu.component.scss'],
})
export class NavMenuComponent implements OnInit {
  @Input() menuItems: any[] = [];
  @Input() currentUser!: UserInfo;

  constructor(private spotifyService: SpotifyService) {}

  ngOnInit(): void {
  }


}
