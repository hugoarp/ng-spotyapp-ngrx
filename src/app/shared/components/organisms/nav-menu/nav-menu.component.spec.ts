import { ComponentFixture, TestBed } from '@angular/core/testing';

import { NavMenuComponent } from './nav-menu.component';

describe('NavMenuComponent', () => {
  let component: NavMenuComponent;
  let fixture: ComponentFixture<NavMenuComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [NavMenuComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(NavMenuComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe hacer match con el snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });
});
