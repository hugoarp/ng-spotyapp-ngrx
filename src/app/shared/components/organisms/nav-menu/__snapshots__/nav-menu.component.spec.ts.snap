// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenuComponent debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <nav
    class="nav-menu"
  >
    <img
      alt="Logo de spotify"
      class="nav-menu__logo"
      src="./assets/img/spotify-logo.svg"
    />
    <section
      class="nav-menu__pages"
    >
      
    </section>
  </nav>
</div>
`;
